control/cli.py

```diff
@@ -1237,6 +1238,8 @@ def ns_del(self, args):
             self.cli.parser.error("--r-megabytes-per-second argument is not allowed for del command")
         if args.w_megabytes_per_second != None:
             self.cli.parser.error("--w-megabytes-per-second argument is not allowed for del command")
+        if args.force:
+            self.cli.parser.error("--force argument is not allowed for del command")
```

Comment:
I am a bit surprised, about "argument is not allowed", it seems that "--force" flag was added to the wrong subparser 
```python
import argparse

def add_namespace(args):
    print(f"Adding namespace with add_flag: {args.add_flag}")

def del_namespace(args):
    print(f"Deleting namespace with del_flag: {args.del_flag}")

def main():
    parser = argparse.ArgumentParser(description="NVMe over Fabrics CLI")

    subparsers = parser.add_subparsers(title="subcommands", dest="command")

    # Subparser for adding namespace
    parser_add = subparsers.add_parser("add", help="Add namespace")
    parser_add.add_argument("--add_flag", help="Flag for adding", required=True)

    # Subparser for deleting namespace
    parser_del = subparsers.add_parser("del", help="Delete namespace")
    parser_del.add_argument("--del_flag", help="Flag for deleting", required=True)

    args = parser.parse_args()

    if args.command == "add":
        add_namespace(args)
    elif args.command == "del":
        del_namespace(args)
    else:
        parser.print_help()

if __name__ == "__main__":
    main()

```

Reply:
@baum there's a limit on the sub-parsers. I couldn't add another level of sub-parsers. Do I had to take care of some of the parsing myself.

Reply:
> @baum there's a limit on the sub-parsers. 

@gbregman, what is the limit, i.e. maximum number of sub-parsers?

> I couldn't add another level of sub-parsers. Do I had to take care of some of the parsing myself.

Why can not add? For now some parsing:
```shell
baum@ceph-nvme-build-server-2 /data/code/ceph-nvmeof (devel) $ grep "argument is not allowed " control/cli.py
            self.cli.parser.error("--level argument is not allowed for disable command")
            self.cli.parser.error("--print argument is not allowed for disable command")
            self.cli.parser.error("--level argument is not allowed for get command")
            self.cli.parser.error("--print argument is not allowed for get command")
            self.cli.parser.error("--force argument is not allowed for add command")
            self.cli.parser.error("--serial-number argument is not allowed for del command")
            self.cli.parser.error("--max-namespaces argument is not allowed for del command")
            self.cli.parser.error("--enable-ha argument is not allowed for del command")
            self.cli.parser.error("--max-namespaces argument is not allowed for list command")
            self.cli.parser.error("--enable-ha argument is not allowed for list command")
            self.cli.parser.error("--force argument is not allowed for list command")
            self.cli.parser.error("--gateway-name argument is not allowed for list command")
            self.cli.parser.error("--traddr argument is not allowed for list command")
            self.cli.parser.error("--adrfam argument is not allowed for list command")
            self.cli.parser.error("--trsvcid argument is not allowed for list command")
            self.cli.parser.error("--host argument is not allowed for list command")
                self.cli.parser.error("--size argument is not allowed for add command when RBD image creation is disabled")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for add command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for add command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for add command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for add command")
            self.cli.parser.error("--size argument is not allowed for del command")
            self.cli.parser.error("--block-size argument is not allowed for del command")
            self.cli.parser.error("--rbd-pool argument is not allowed for del command")
            self.cli.parser.error("--rbd-image argument is not allowed for del command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for del command")
            self.cli.parser.error("--load-balancing-group argument is not allowed for del command")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for del command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for del command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for del command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for del command")
            self.cli.parser.error("--block-size argument is not allowed for resize command")
            self.cli.parser.error("--rbd-pool argument is not allowed for resize command")
            self.cli.parser.error("--rbd-image argument is not allowed for resize command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for resize command")
            self.cli.parser.error("--load-balancing-group argument is not allowed for resize command")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for resize command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for resize command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for resize command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for resize command")
            self.cli.parser.error("--size argument is not allowed for list command")
            self.cli.parser.error("--block-size argument is not allowed for list command")
            self.cli.parser.error("--rbd-pool argument is not allowed for list command")
            self.cli.parser.error("--rbd-image argument is not allowed for list command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for list command")
            self.cli.parser.error("--load-balancing-group argument is not allowed for list command")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for list command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for list command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for list command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for list command")
            self.cli.parser.error("--size argument is not allowed for get_io_stats command")
            self.cli.parser.error("--block-size argument is not allowed for get_io_stats command")
            self.cli.parser.error("--rbd-pool argument is not allowed for get_io_stats command")
            self.cli.parser.error("--rbd-image argument is not allowed for get_io_stats command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for get_io_stats command")
            self.cli.parser.error("--load-balancing-group argument is not allowed for get_io_stats command")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for get_io_stats command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for get_io_stats command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for get_io_stats command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for get_io_stats command")
            self.cli.parser.error("--size argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--block-size argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--rbd-pool argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--rbd-image argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--rw-ios-per-second argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--r-megabytes-per-second argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--w-megabytes-per-second argument is not allowed for change_load_balancing_group command")
            self.cli.parser.error("--load-balancing-group argument is not allowed for set_qos command")
            self.cli.parser.error("--size argument is not allowed for set_qos command")
            self.cli.parser.error("--block-size argument is not allowed for set_qos command")
            self.cli.parser.error("--rbd-pool argument is not allowed for set_qos command")
            self.cli.parser.error("--rbd-image argument is not allowed for set_qos command")
            self.cli.parser.error("--rbd-create-image argument is not allowed for set_qos command")
baum@ceph-nvme-build-server-2 /data/code/ceph-nvmeof (devel) $ grep "argument is not allowed " control/cli.py  | wc -l
75
```

Reply:
@baum I remember looking into it and couldn't do it. I can try again, in any case it has nothing to do with this issue. It needs to be done under a separate issue.

Reply:
Sounds good, see #432

Reply:
I couldn't add a second level of sub parsers, it fails. As there is already one sub-parser we use I couldn't add another one below it.